preprocessing_snippet.R

```diff
@@ -0,0 +1,18 @@
+# create function to normalize counts in deseq object
+# assume rowRanges is a non-empty GRangesList and size factors were already computed (ex: using estimateSizeFactors())
+normCounts <- function(se) {
+  gene_length <- sum(width(rowRanges(se)))
+  counts_norm <- apply(counts(se, normalized = TRUE), 2, 
+                       function(x) {
+                         1e3 * x / gene_length
+                       })
+  assays(se)$counts_norm <- counts_norm
+  return(se)
+}
+
+
+# prepare collection of all exons for genes in gencode 
+# only want exons, not introns, because introns are not expressed
+genes <- levels(as.factor(gencode$gene_name))
+exons <- gencode[gencode$type == 'exon']
```

Comment:
this will need to be updated as well.  as we are doing `exons <- gencode[(gencode$gene_type == 'protein_coding') & (gencode$type == 'exon')]` now

Reply:
Fixed - But, I left the old code as well because the original analyses were done with all genes, not just the protein coding genes.